mutters-core/src/main/java/com/rabidgremlin/mutters/core/MatcherScores.java

```diff
@@ -97,4 +97,67 @@ public OptionalDouble getBestScore()
     }
   }
 
+  /**
+   * Combines two {@link MatcherScores}.
+   * <p>
+   * If there are duplicate scores for a given intent, takes the highest score.
+   *
+   * @param firstScores  first scores
+   * @param secondScores second scores
+   * @return matcher scores consisting of scores from the first and second scores
+   */
+  public static MatcherScores combine(MatcherScores firstScores, MatcherScores secondScores)
+  {
+    // only combine if necessary
+    if (firstScores.isEmpty())
+    {
+      return secondScores;
+    }
+    else if (secondScores.isEmpty())
+    {
+      return firstScores;
+    }
+
+    // flatten [Double, Set[String]] into [String, Double]
+    // taking the highest score for any duplicate intent scores
+    Map<String, Double> flattenedScores = Stream
```

Comment:
Taking the highest score works but will skew the scoring. Also one matcher may be wildly out in its prediction so highest score is not necessary a good score :(